aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -120,5 +122,92 @@ Tensor prod(const Tensor &self, int64_t dim_, bool keepdim) {
 }
 
 // \DIM REDUCE ################################################################
+
+// MULTI DIM REDUCE ###########################################################
+
+template <Tensor (reduce_1)(const Tensor &, int64_t, bool)>
+inline Tensor reduce_multi(const Tensor &self, IntList dims_, bool keepdim) {
+  if (dims_.size() == 1) {
+    return reduce_1(self, dims_[0], keepdim);
+  }
+  if (dims_.size() == 0) {
+    return self;
+  }
+  size_t ndims = self.dim();
+  AT_ASSERT(ndims <= (int64_t) dim_bitset_size, "tensor dimension must be <= %zu for multiple dims", dim_bitset_size);
+  std::bitset<dim_bitset_size> seen;
+  Tensor result = self;
+  for (size_t i = 0; i < dims_.size(); i++) {
+    size_t dim = maybe_wrap_dim(dims_[i], ndims);
+    if (seen[dim])
+      AT_ERROR("repeated dim");
+    seen[dim] = true;
+    result = reduce_1(result, dim, true);
+  }
```

Comment:
This is duplicated code, and it's better to avoid it. Let's just use the function to create the bitset and validate the arg, and then just iterate over dims again.